diff --git a/node_modules/@fengweichong/react-native-gzip/android/build.gradle b/node_modules/@fengweichong/react-native-gzip/android/build.gradle
index c2e330f..f28740e 100644
--- a/node_modules/@fengweichong/react-native-gzip/android/build.gradle
+++ b/node_modules/@fengweichong/react-native-gzip/android/build.gradle
@@ -73,7 +73,7 @@ repositories {
 dependencies {
     //noinspection GradleDynamicVersion
     implementation 'com.facebook.react:react-native:+'  // From node_modules
-    implementation 'commons-io:commons-io:2.6'
+    implementation 'commons-io:commons-io:2.5'
     implementation 'org.apache.commons:commons-compress:1.1'
 }
 
