diff --git a/node_modules/@mythologi/react-native-receive-sharing-intent/src/ReceiveSharingIntent.ts b/node_modules/@mythologi/react-native-receive-sharing-intent/src/ReceiveSharingIntent.ts
index 2ad1da2..3381534 100644
--- a/node_modules/@mythologi/react-native-receive-sharing-intent/src/ReceiveSharingIntent.ts
+++ b/node_modules/@mythologi/react-native-receive-sharing-intent/src/ReceiveSharingIntent.ts
@@ -33,17 +33,12 @@ class ReceiveSharingIntentModule implements IReceiveSharingIntent {
         }
       });
     } else {
-      AppState.addEventListener('change', (status: string) => {
-        if (status === 'active' && !this.isClear) {
-          this.getFileNames(handler, errorHandler, '');
-        }
-      });
-      if (!this.isClear) this.getFileNames(handler, errorHandler, '');
+      this.getFileNames(handler,errorHandler, "");
     }
   }
 
   clearReceivedFiles() {
-    this.isClear = true;
+    ReceiveSharingIntent.clearFileNames()
   }
 
   protected getFileNames(
diff --git a/node_modules/@mythologi/react-native-receive-sharing-intent/src/utils.ts b/node_modules/@mythologi/react-native-receive-sharing-intent/src/utils.ts
index 60e638e..62540db 100644
--- a/node_modules/@mythologi/react-native-receive-sharing-intent/src/utils.ts
+++ b/node_modules/@mythologi/react-native-receive-sharing-intent/src/utils.ts
@@ -61,7 +61,7 @@ class Utils implements IUtils {
     const type = Object.entries(MimeTypes).find(
       (mime) => mime[0] === extension
     );
-    if (type) return type[0];
+    if (type) return type[1];
     return '';
   };
 }
