diff --git a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
index e25c07c..bd1c08c 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
+++ b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
@@ -424,6 +424,8 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
 
       NSArray<NSString*> *const assetMediaSubtypesLabel = [self mediaSubTypeLabelsForAsset:asset];
 
+      NSArray<NSString*> *const albums = [self getAlbumsForAsset:asset];
+
       if (includeFileExtension) {
         NSString *name = [asset valueForKey:@"filename"];
         NSString *extension = [name pathExtension];
@@ -436,7 +438,7 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
         @"node": @{
           @"type": assetMediaTypeLabel, // TODO: switch to mimeType?
           @"subTypes":assetMediaSubtypesLabel,
-          @"group_name": currentCollectionName,
+          @"group_name": albums,
           @"image": @{
               @"uri": uri,
               @"extension": (includeFileExtension ? fileExtension : [NSNull null]),
@@ -715,6 +717,18 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
     return mediaSubTypeLabels;
 }
 
+- (NSArray<NSString *> *)getAlbumsForAsset:(PHAsset *)asset {
+    NSMutableArray<NSString *> *albumTitles = [NSMutableArray array];
+
+    PHFetchResult<PHAssetCollection *> *collections = [PHAssetCollection fetchAssetCollectionsContainingAsset:asset withType:PHAssetCollectionTypeAlbum options:nil];
+
+    for (PHAssetCollection *collection in collections) {
+        [albumTitles addObject:collection.localizedTitle];
+    }
+
+    return [albumTitles copy];
+}
+
 static void checkPhotoLibraryConfig()
 {
 #if RCT_DEV
diff --git a/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/CameraRoll.d.ts b/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/CameraRoll.d.ts
index 45ac596..bb573e1 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/CameraRoll.d.ts
+++ b/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/CameraRoll.d.ts
@@ -51,7 +51,7 @@ export declare type PhotoIdentifier = {
     node: {
         type: string;
         subTypes: SubTypes;
-        group_name: string;
+        group_name: string | string [];
         image: {
             filename: string | null;
             filepath: string | null;
diff --git a/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/NativeCameraRollModule.d.ts b/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/NativeCameraRollModule.d.ts
index 26346db..fe40652 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/NativeCameraRollModule.d.ts
+++ b/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/NativeCameraRollModule.d.ts
@@ -8,7 +8,7 @@ declare type PhotoIdentifier = {
     node: {
         type: string;
         subTypes: SubTypes;
-        group_name: string;
+        group_name: string | string[];
         image: {
             filename: string | null;
             filepath: string | null;
diff --git a/node_modules/@react-native-camera-roll/camera-roll/src/CameraRoll.ts b/node_modules/@react-native-camera-roll/camera-roll/src/CameraRoll.ts
index 871e84c..cb1ad1e 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/src/CameraRoll.ts
+++ b/node_modules/@react-native-camera-roll/camera-roll/src/CameraRoll.ts
@@ -111,7 +111,7 @@ export type PhotoIdentifier = {
   node: {
     type: string;
     subTypes: SubTypes;
-    group_name: string;
+    group_name: string | string[];
     image: {
       filename: string | null;
       filepath: string | null;
diff --git a/node_modules/@react-native-camera-roll/camera-roll/src/NativeCameraRollModule.ts b/node_modules/@react-native-camera-roll/camera-roll/src/NativeCameraRollModule.ts
index c4f321f..d690cca 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/src/NativeCameraRollModule.ts
+++ b/node_modules/@react-native-camera-roll/camera-roll/src/NativeCameraRollModule.ts
@@ -22,7 +22,7 @@ type PhotoIdentifier = {
   node: {
     type: string;
     subTypes: SubTypes;
-    group_name: string;
+    group_name: string | string [];
     image: {
       filename: string | null;
       filepath: string | null;
