diff --git a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
index 76b27ad..58555ea 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
+++ b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
@@ -311,6 +311,8 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
   NSArray<NSString *> *const mimeTypes = [RCTConvert NSStringArray:params[@"mimeTypes"]];
   NSArray<NSString *> *const include = [RCTConvert NSStringArray:params[@"include"]];

+  BOOL __block includeSharedAlbums = [params[@"includeSharedAlbums"] boolValue];
+
   BOOL __block includeFilename = [include indexOfObject:@"filename"] != NSNotFound;
   BOOL __block includeFileSize = [include indexOfObject:@"fileSize"] != NSNotFound;
   BOOL __block includeFileExtension = [include indexOfObject:@"fileExtension"] != NSNotFound;
@@ -341,6 +343,10 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
   }
   assetFetchOptions.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"creationDate" ascending:NO]];

+  if (includeSharedAlbums) {
+    assetFetchOptions.includeAssetSourceTypes = PHAssetSourceTypeUserLibrary | PHAssetSourceTypeCloudShared;
+  }
+
   BOOL __block foundAfter = NO;
   BOOL __block hasNextPage = NO;
   BOOL __block resolvedPromise = NO;
diff --git a/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/CameraRoll.d.ts b/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/CameraRoll.d.ts
index bb573e1..892a873 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/CameraRoll.d.ts
+++ b/node_modules/@react-native-camera-roll/camera-roll/lib/typescript/CameraRoll.d.ts
@@ -25,6 +25,10 @@ export declare type GetPhotosParams = {
      * titles.
      */
     groupName?: string;
+    /**
+     * Include assets originating from an iCloud Shared Album. iOS only.
+     */
+    includeSharedAlbums?: boolean;
     /**
      * Specifies filter on asset type
      */
diff --git a/node_modules/@react-native-camera-roll/camera-roll/src/CameraRoll.ts b/node_modules/@react-native-camera-roll/camera-roll/src/CameraRoll.ts
index cb1ad1e..cf471a1 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/src/CameraRoll.ts
+++ b/node_modules/@react-native-camera-roll/camera-roll/src/CameraRoll.ts
@@ -80,6 +80,11 @@ export type GetPhotosParams = {
    */
   groupName?: string;

+  /**
+   * Include assets originating from an iCloud Shared Album. iOS only.
+   */
+  includeSharedAlbums?: boolean;
+
   /**
    * Specifies filter on asset type
    */
