diff --git a/node_modules/react-native-file-viewer/android/src/main/AndroidManifest.xml b/node_modules/react-native-file-viewer/android/src/main/AndroidManifest.xml
index 8166e4c..4be50e7 100644
--- a/node_modules/react-native-file-viewer/android/src/main/AndroidManifest.xml
+++ b/node_modules/react-native-file-viewer/android/src/main/AndroidManifest.xml
@@ -4,7 +4,7 @@
   package="com.vinzscam.reactnativefileviewer"
   >
       <application>
-        <provider android:name="com.vinzscam.reactnativefileviewer.FileProvider" android:authorities="${applicationId}.provider" android:exported="false" android:grantUriPermissions="true">
+        <provider android:name="com.vinzscam.reactnativefileviewer.FileProvider" android:authorities="${applicationId}.reactnativefileviewer.provider" android:exported="false" android:grantUriPermissions="true">
               <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/file_viewer_provider_paths" />
         </provider>
       </application>
diff --git a/node_modules/react-native-file-viewer/android/src/main/java/com/vinzscam/reactnativefileviewer/RNFileViewerModule.java b/node_modules/react-native-file-viewer/android/src/main/java/com/vinzscam/reactnativefileviewer/RNFileViewerModule.java
index 93f7881..9293eab 100644
--- a/node_modules/react-native-file-viewer/android/src/main/java/com/vinzscam/reactnativefileviewer/RNFileViewerModule.java
+++ b/node_modules/react-native-file-viewer/android/src/main/java/com/vinzscam/reactnativefileviewer/RNFileViewerModule.java
@@ -59,7 +59,7 @@ public class RNFileViewerModule extends ReactContextBaseJavaModule {
       }
       try {
         final String packageName = currentActivity.getPackageName();
-        final String authority = new StringBuilder(packageName).append(".provider").toString();
+        final String authority = new StringBuilder(packageName).append(".reactnativefileviewer.provider").toString();
         contentUri = FileProvider.getUriForFile(currentActivity, authority, newFile);
       }
       catch(IllegalArgumentException e) {
diff --git a/node_modules/react-native-file-viewer/ios/RNFileViewerManager.m b/node_modules/react-native-file-viewer/ios/RNFileViewerManager.m
index 8149cac..6ea97af 100644
--- a/node_modules/react-native-file-viewer/ios/RNFileViewerManager.m
+++ b/node_modules/react-native-file-viewer/ios/RNFileViewerManager.m
@@ -102,6 +102,12 @@ - (void)previewControllerDidDismiss:(CustomQLViewController *)controller {
     [self sendEventWithName:DISMISS_EVENT body: @{@"id": controller.invocation}];
 }
 
+- (void)dismissView:(id)sender {
+    UIViewController* controller = [RNFileViewer topViewController];
+    [self sendEventWithName:DISMISS_EVENT body: @{@"id": ((CustomQLViewController*)controller).invocation}];
+    [[RNFileViewer topViewController] dismissViewControllerAnimated:YES completion:nil];
+}
+
 RCT_EXPORT_MODULE()
 
 - (NSArray<NSString *> *)supportedEvents {
@@ -117,8 +123,17 @@ - (void)previewControllerDidDismiss:(CustomQLViewController *)controller {
     QLPreviewController *controller = [[CustomQLViewController alloc] initWithFile:file identifier:invocationId];
     controller.delegate = self;
 
+    if (@available(iOS 13.0, *)) {
+        [controller setModalInPresentation: true];
+    }
+
+    UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:controller];
+    controller.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self action:@selector(dismissView:)];
+    controller.navigationController.navigationBar.translucent = NO;
+    controller.edgesForExtendedLayout = UIRectEdgeNone;
+
     typeof(self) __weak weakSelf = self;
-    [[RNFileViewer topViewController] presentViewController:controller animated:YES completion:^{
+    [[RNFileViewer topViewController] presentViewController:navigationController animated:YES completion:^{
         [weakSelf sendEventWithName:OPEN_EVENT body: @{@"id": invocationId}];
     }];
 }
