diff --git a/node_modules/react-native-file-logger/android/build.gradle b/node_modules/react-native-file-logger/android/build.gradle
index 55f0b45..6c083ca 100644
--- a/node_modules/react-native-file-logger/android/build.gradle
+++ b/node_modules/react-native-file-logger/android/build.gradle
@@ -97,8 +97,8 @@ repositories {
 dependencies {
     //noinspection GradleDynamicVersion
     implementation "com.facebook.react:react-native:+"
-    implementation 'org.slf4j:slf4j-api:1.7.33'
-    implementation 'com.github.tony19:logback-android:2.0.0'
+    implementation 'org.slf4j:slf4j-api:2.0.7'
+    implementation 'com.github.tony19:logback-android:3.0.0'
 }
 
 if (isNewArchitectureEnabled()) {
