diff --git a/node_modules/react-native-performance-flipper-reporter/src/index.js b/node_modules/react-native-performance-flipper-reporter/src/index.js
index e99d526..bf1f6f1 100644
--- a/node_modules/react-native-performance-flipper-reporter/src/index.js
+++ b/node_modules/react-native-performance-flipper-reporter/src/index.js
@@ -163,7 +163,7 @@ export function setupDefaultFlipperReporter() {
               name: entry.name,
               startTime: entry.startTime,
               duration: entry.duration,
-              category: 'App',
+              category: entry.detail?.category || 'App',
             }))
           );
         },
